core/market/decentralized/tests/test_initial_proposal.rs

```diff
@@ -155,62 +203,43 @@ mod tests {
             .add_market_instance("Node-1")
             .await?;
 
-        let node1 = network.get_node("Node-1");
-        let market1: Arc<MarketService> = network.get_market("Node-1");
+        let market1 = network.get_market("Node-1");
         let identity1 = network.get_default_id("Node-1");
-
-        let (_offer_id, demand_id) = node1
-            .inject_proposal(&example_offer(), &example_demand())
+        let demand_id = market1
+            .subscribe_demand(&sample_demand(), &identity1)
             .await?;
+        market1.subscribe_offer(&sample_offer(), &identity1).await?;
 
         // We should reject calls with negative maxEvents.
-        match market1
-            .requestor_engine
-            .query_events(&demand_id, 0.0, Some(-5))
-            .await
-        {
-            Err(QueryEventsError::InvalidMaxEvents(value)) => {
-                assert_eq!(value, -5);
-            }
-            _ => panic!("Negative maxEvents - expected error"),
-        };
 
-        // Negative timeout should be treated as immediate checking events and return.
-        let events = tokio::time::timeout(
-            Duration::from_millis(20),
-            market1
-                .requestor_engine
-                .query_events(&demand_id, -5.0, None),
-        )
-        .await??;
```

Comment:
Why did you removed this case?

Reply:
I've removed because we need to wait a bit for resolver to process subscriptions. Haven't considered to obviously wait before calling it :) restoring